src/osx/cocoa/dataview.mm

```diff
@@ -820,36 +813,34 @@ -(NSDragOperation) setupAndCallDataViewEvents:(wxEventType)eventType dropInfo:(i
             else
                 indexDraggedItem = noOfDraggedItems;
 
-            // clean-up:
             delete dataObjects;
+            dataObjects = NULL;
         }
     }
     else
     {
-        // needed to convert internally used UTF-16 representation to a UTF-8
-        // representation
-        CFDataRef              osxData;
-        wxDataObjectComposite* dataObjects   (new wxDataObjectComposite());
-        wxTextDataObject*      textDataObject(new wxTextDataObject());
+        dataObjects = new wxDataObjectComposite();
 
-        osxData = ::CFStringCreateExternalRepresentation(kCFAllocatorDefault,(CFStringRef)[pasteboard stringForType:NSStringPboardType],
-#if defined(wxNEEDS_UTF16_FOR_TEXT_DATAOBJ)
-                                                         kCFStringEncodingUTF16,
-#else
-                                                         kCFStringEncodingUTF8,
-#endif
-                                                         32);
-        if (textDataObject->SetData(::CFDataGetLength(osxData),::CFDataGetBytePtr(osxData)))
-            dataObjects->Add(textDataObject);
-        else
-            delete textDataObject;
-        // send event if data could be copied:
+        if ([bestType compare:NSFilenamesPboardType] == NSOrderedSame)
+        {
+            dataObject = new wxFileDataObject;
+        }
+        else if ([bestType compare:NSStringPboardType] == NSOrderedSame)
+        {
+            dataObject = new wxTextDataObject;
+        }
+        // TODO check for other formats here?
```

Comment:
I covered only `NSFilenamesPboardType` and `NSStringPboardType` here. Not sure if the rest of the types specified in the `EnableDropTarget` will work.

Reply:
if you move the 'generic' EnableDropTarget as suggested below, then you can make the else branch start like this

```
    else
    {
        dataObjects = new wxDataObjectComposite();
        wxDataFormat wxFormat((wxDataFormat::NativeFormat)bestType);
        switch (wxFormat.GetType() )
        {
            case wxDF_FILENAME :
                dataObject = new wxFileDataObject;
                break;
            case wxDF_TEXT :
            case wxDF_UNICODETEXT:
                dataObject = new wxFileDataObject;
                break;
            default:
                break;
        }

```

That way all modern UTI types (the PboardTypes you were using will be deprecated) are supported automatically, and you could move this code again to a superclass ...